### PR TITLE
Build Funder Email Confirmation with Feature Bounty 

### LINKED ISSUE #3651: Build Funder Email Confirmation with Feature Bounty
When I found a bounty, I don't receive an email confirmation that my bounty was successfully funded.

### Description

Build the email so that funders know their bounty has been funded.

### Definition of Done

- [ ] This is a front-end and backend task, at completion this should be a working email.
- [ ] The email should look like the attached design
- [ ] The email should support be able to be dynamic and not display the featured bounty section when he funder does not pay extra for featured bounty.
- [ ] The featured bounty section should show when the funder has selected and paid for the featured bounty on the funder form.
- [ ] Email should be responsive
- [ ] Review with Gitcoin Core team members.

Design:
bounty success email

### PR DESCRIPTION
**Pull request by iamonuwa**
##### Checklist

- [x] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [x] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [x] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Affected core subsystem(s)

UI
##### Refers/Fixes

Fixes #3651
###### Contributor

- [x] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Tested all changes **locally**
- [x] Verified existing functionality
- [x] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

File: app/assets/v2/js/grants/fund.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ $(document).ready(function() {
           }).on('transactionHash', function(transactionHash) {
             $('#sub_new_approve_tx_id').val(transactionHash);
             const linkURL = etherscan_tx_url(transactionHash);
-            var token_address = $('#js-token').length ? $('#js-token').val() : $("#sub_token_address").val();
+            var token_address = $('#js-token').length ? $('#js-token').val() : $('#sub_token_address').val();
```
Comment (danlipert) [inline app/assets/v2/js/grants/fund.js]: Why was this changed? Just to clean up the style?

File: app/dashboard/helpers.py [hunk 2]
```diff
@@ -450,12 +450,17 @@ def create_new_bounty(old_bounties, bounty_payload, bounty_details, bounty_id):
                 latest_old_bounty_dict['bounty_reserved_for_user'] = Profile.objects.get(pk=latest_old_bounty_dict['bounty_reserved_for_user'])
             bounty_kwargs.update(latest_old_bounty_dict)
 
+
```
Comment (danlipert) [inline app/dashboard/helpers.py]: Please remove this extra whitespace 👍

File: app/dashboard/helpers.py [hunk 3]
```diff
@@ -450,12 +450,17 @@ def create_new_bounty(old_bounties, bounty_payload, bounty_details, bounty_id):
...
         try:
             new_bounty = Bounty.objects.create(**bounty_kwargs)
             new_bounty.fetch_issue_item()
             try:
                 issue_kwargs = get_url_dict(new_bounty.github_url)
                 new_bounty.github_issue_details = get_gh_issue_details(**issue_kwargs)
+
+                # If bounty is_featured flag is on
```
Comment (danlipert) [inline app/dashboard/helpers.py]: I think this comment isn't needed since the code is so readable, what do you think?

File: app/dashboard/views.py [hunk 4]
```diff
@@ -1423,7 +1423,6 @@ def terms(request):
     }
     return TemplateResponse(request, 'legal/terms.html', context)
 
-
```
Comment (danlipert) [inline app/dashboard/views.py]: Is this a style fix?

File: app/assets/v2/js/grants/fund.js (hunk 1 above)
Comment (iamonuwa) [reply app/assets/v2/js/grants/fund.js]: I ran eslint:fix and it pointed it out

File: app/dashboard/helpers.py (hunk 3 above)
Comment (iamonuwa) [reply app/dashboard/helpers.py]: I'll remove the comment

**Comment by danlipert** [comment]
@iamonuwa Great! Thanks for that - let me know if you need help with the testing, I think there are some good examples of testing emails in the test cases already so it should be mostly copy and paste 👍

File: app/dashboard/tests/test_dashboard_helpers.py [hunk 5]
```diff
@@ -50,3 +55,28 @@ def test_amount(self):
...
+            accepted=False,
+            expires_date=datetime(2008, 11, 30, tzinfo=pytz.UTC),
+            idx_project_length=5,
+            project_length='Months',
+            bounty_type='Feature',
+            experience_level='Intermediate',
+            raw_data={},
+            is_featured=True
+        )
+        assert mock_featured_funded_bounty().call_count == 0
```
Comment (danlipert) [inline app/dashboard/tests/test_dashboard_helpers.py]: So in this test, we are confirming that the featured bounty email is sent - but here, the `call_count` is 0, indicating that an email was not sent. This test needs to assert that `call_count == 1`, which shows that an email would have been sent.

Comment (iamonuwa) [reply app/dashboard/tests/test_dashboard_helpers.py]: Got that. I'll refactor this now.

**Comment by iamonuwa** [comment]
@danlipert please take a look. Test still fails though

File: app/dashboard/tests/test_dashboard_helpers.py [hunk 6]
```diff
@@ -50,3 +55,102 @@ def test_amount(self):
...
+            "additional_funding_summary": {},
+            "funding_organisation": "",
+            "paid": [
+                "danlipert"
+            ],
+            "admin_override_suspend_auto_approval":True,
+            "reserved_for_user_handle": "",
+            "is_featured":True
+        },
+        process_bounty_details(bounty_payload)
```
Comment (danlipert) [inline app/dashboard/tests/test_dashboard_helpers.py]: @iamonuwa try calling `create_new_bounty` here, either directly or indirectly

Comment (iamonuwa) [reply app/dashboard/tests/test_dashboard_helpers.py]: `create_new_bounty` is not called directly in the code. Its called through `process_bounty_details`

Comment (iamonuwa) [reply app/dashboard/tests/test_dashboard_helpers.py]: The additional parameters for `create_new_bounty` is generated inside the `process_bounty_details`

Comment (iamonuwa) [reply app/dashboard/tests/test_dashboard_helpers.py]: @danlipert can you confirm?

**Comment by iamonuwa** [comment]
@danlipert @octavioamu @PixelantDesign am closing this PR. I deleted my gitcoin web repo unintentionally.

Here is the new repo with the tests passing
